src/components/blending-opacity.js

```diff
@@ -0,0 +1,145 @@
+  },
+
+  update: function (oldData) {
+    // Only update if the data has actually changed
+    if (
+      oldData.tintColor !== this.data.tintColor ||
+      oldData.opacity !== this.data.opacity ||
+      oldData.blendMode !== this.data.blendMode
+    ) {
+      this.updateMaterials();
+    }
+  },
+
+  remove: function () {
+    this.el.removeEventListener('model-loaded', this.onModelLoaded);
+  },
+
+  onModelLoaded: function () {
+    this.updateMaterials();
+  },
+
+  setCustomBlendMode: function (material, blendMode) {
+    material.blending = THREE.CustomBlending;
+
+    switch (blendMode) {
+      case 'Screen':
+        material.blendEquation = THREE.AddEquation;
+        material.blendSrc = THREE.OneFactor;
+        material.blendDst = THREE.OneMinusSrcColorFactor;
+        break;
+      case 'Overlay':
+        material.blendEquation = THREE.AddEquation;
+        material.blendSrc = THREE.OneFactor;
+        material.blendDst = THREE.OneMinusSrcAlphaFactor;
+        break;
+      case 'Lighten':
+        material.blendEquation = THREE.MaxEquation;
+        material.blendSrc = THREE.OneFactor;
+        material.blendDst = THREE.OneFactor;
+        break;
+      case 'Darken':
+        material.blendEquation = THREE.MinEquation;
+        material.blendSrc = THREE.OneFactor;
+        material.blendDst = THREE.OneFactor;
+        break;
+      case 'ColorDodge':
+        material.blendEquation = THREE.AddEquation;
+        material.blendSrc = THREE.OneFactor;
+        material.blendDst = THREE.OneMinusSrcColorFactor;
+        break;
+      case 'ColorBurn':
+        material.blendEquation = THREE.AddEquation;
+        material.blendSrc = THREE.OneMinusDstColorFactor;
+        material.blendDst = THREE.OneFactor;
+        break;
+    }
+  },
+
+  updateMaterials: function () {
+    const opacity = this.data.opacity;
+    const blendMode = this.data.blendMode;
+
+    this.el.object3D.traverse((node) => {
+      if (node.type === 'Mesh') {
+        const materials = Array.isArray(node.material)
+          ? node.material
+          : [node.material];
+
+        materials.forEach((material) => {
+          material.transparent = opacity < 1.0;
+          material.opacity = opacity;
+
+          // Set blending mode
+          if (
+            blendMode === 'Normal' ||
+            blendMode === 'Additive' ||
+            blendMode === 'Subtract' ||
+            blendMode === 'Multiply'
+          ) {
+            material.blending = this.blendModes[blendMode];
+          } else {
+            this.setCustomBlendMode(material, blendMode);
+          }
+
+          material.needsUpdate = true;
+        });
+      }
+    });
+  },
+
+  tick: function () {
+    // Only update materials if there are changes
+    // if (this.el.getAttribute('tiles-material-change').needsUpdate) {
+    //   this.updateMaterials();
+    //   this.el.getAttribute('tiles-material-change').needsUpdate = false;
+    // }
+    this.updateMaterials();
```

Comment:
That's expensive to keep doing a traverse on each tick, also it's creating an array each time `[node.material]`.

Reply:
You're right, in obb-clipping this is throttled, I can do that again.

Reply:
Unfortunately the result looks very bad if throttled, materials appear to "flicker" as they wait for the throttle to be adjusted to a different blending.

For now I'll keep it, but there are 2 options to fix:
1. hook into the material post processing event in the existing 3d tiles library
2. consider switching to better maintained library instead of optimizing this one, hook into its material post processing event https://github.com/NASA-AMMOS/3DTilesRendererJS?tab=readme-ov-file#custom-material (blocked by a-frame / three.js upgrade -- needs v166)